agent/loop.py

```
@@ -657,6 +666,15 @@ async def _generate_plan(
         messages = build_planning_prompt(instruction, history, user_memories=user_memories)
         # ── Harness enrichment: inject available tools + skills into planner ───
         self._inject_enrichment(messages)
+        # ── Learning loop: surface lessons from recent failed runs so the
+        # planner avoids known failure modes (agent/lessons.py).
+        try:
+            from agent.lessons import recent_lessons_block
+            _lessons = recent_lessons_block()
+            if _lessons and messages and messages[0].get("role") == "system":
+                messages[0]["content"] = f"{messages[0]['content']}\n\n{_lessons}"
+        except Exception:
+            pass
```

Ruff (0.15.20)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)